darshan-util/pydarshan/darshan/backend/cffi_backend.py

```diff
@@ -565,29 +565,31 @@ def log_get_dxt_record(log, mod_name, reads=True, writes=True, dtype='dict'):
 
     size_of = ffi.sizeof("struct dxt_file_record")
     segments = ffi.cast("struct segment_info *", buf[0] + size_of  )
-
+    arr_write = np.recarray(wcnt, dtype=[("offset", int),
+                                         ("length", int),
+                                         ("start_time", float),
+                                         ("end_time", float)])
+    arr_read = np.recarray(rcnt, dtype=[("offset", int),
+                                        ("length", int),
+                                        ("start_time", float),
+                                        ("end_time", float)])
```

Comment:
I really liked the idea of using `recarray` because it kind of retains the original metadata in the structured dtype, but it seems that this carries a huge performance overhead against conventional NumPy `float64` arrays.

I believe the test suite actually didn't complain too much about storing all the data as say `float64`, though this would be something we should consider (is the int->float casting safe in general, etc.).

Reply:
We may also want to try `pandas` DataFrame as well perhaps. There's a bit of a design issue where we have `to_numpy()` and `to_df()` methods, where one of those may become redundant depending on what we decide here, and `to_dict()` is likely a bit idea based on the memory profiling.